rtx_timer_test.go

```diff
@@ -310,7 +311,7 @@ func TestRtxTimer(t *testing.T) {
 		<-doneCh
 
 		assert.True(t, rt.isRunning(), "should still be running")
-		assert.Equal(t, 6, nCbs, "must have been called 6 times")
+		assert.Equal(t, int32(6), atomic.LoadInt32(&nCbs), "should be called 6 times")
```

Comment:
As the old code caused data race with `nCbs++`, I think using atomic just replace data race error by `6 not equals to nCbs`.

Reply:
Thanks!
Btw, could you please point me to the problem? I only saw nCbs error and did not notice other problems you are takling about?

Reply:
As the race detector detected a data race, `nCbs++` and `if nCbs != expectedNCbs` in `assert.Equal` were executed concurrently. (These two memory access operations were reachable at same time from two routines.)
It should mean that `nCbs` potentially take two different values at `assert.Equal` according to the CPU load and task switch timing.

Reply:
Ah.. I see. Yes, those two events (1: nCbs++ and 2: assert.Equal) occur in two different routines but from the same cause, `onRTO` tho.  From what I can see in this particular test when onRTO callback is made, nCbs++ happens first, followed immediately by assert.Equal.  When nRtos = 4, the timeout interval would be 320msec.  If there was a delay of more than 320 msec between nCbs++ and assert.Equal, then yes, nCbs could be 7 but I think it won't happen.

Reply:
Ah...  I guess you were talking about L.206, not L.313.  Let me take a closer look at the test case.